server/models/user.js

```diff
@@ -115,8 +115,44 @@ class User extends MongoModels {
     this.findOne(query, callback);
   }
 
+  static lowestRole(roles){
+
+    if (Array.isArray(roles)){
+      const hm = {};
+      roles.forEach((role) => {
+
+        hm[role] = true;
+      });
+      roles = hm;
+    }
+    if (roles.analyst) {
+      return 1;
+    }
+    else if (roles.clinician) {
+      return 2;
+    }
+    else if (roles.researcher) {
+      return 3;
+    }
+    else if (roles.admin) {
+      return 4;
+    }
+    else if (roles.root) {
+      return 5;
+    }
+    return 0;
+  }
+
   static highestRole(roles) {
 
+    if (Array.isArray(roles)) {//if array, convert to hashtable
+      const hm = {};
+      roles.forEach((role) => {
+
+        hm[role] = true;
+      });
+      roles = hm;
+    }
```

Comment:
refactor your implementation of highest role! I still see hard coded roles in it! you should look at the role object in config and get the highest access level of the user roles by looking up that role object in config file.